android-test/src/androidTest/java/okhttp/android/test/OffMainThread.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2025 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp.android.test
+
+import android.os.Looper
+import okhttp3.Call
+import okhttp3.Request
+import okhttp3.Response
+
+object OffMainThread : Call.Decorator {
+  override fun newCall(
+    chain: Call.Factory,
+    request: Request,
+  ): Call = StrictModeCall(chain.newCall(request))
+
+  private class StrictModeCall(
+    private val delegate: Call,
+  ) : Call by delegate {
+    override fun execute(): Response {
+      if (Looper.getMainLooper().isCurrentThread) {
+        throw IllegalStateException("Network on main thread")
```

Comment:
Not a strong opinion, but I've seen `IllegalThreadStateException` used in our codebase.. though idk that's meant to "indicate that a thread is not in an appropriate state for the requested operation" so not sure. And weird that exception extends IllegalArgumentException